lib/components/app/desktop-nav.tsx

```diff
@@ -70,10 +69,15 @@ const DesktopNav = ({ otpConfig }: otpConfigType) => {
 
       <Navbar.Collapse>
         <Nav pullRight>
-          {/* Don't render locale selector if we have only one language defined */}
-          {localeSelectorEnabled &&
-            configLanguages &&
-            Object.keys(configLanguages).length > 1 && (
+          {configLanguages &&
+            // Avoid rendering the LocaleSelector for configs without languages
+            !Object.keys(configLanguages).find(
+              (key) => key !== 'allLanguages' && !configLanguages[key].name
+            ) &&
+            // Ensure that > 1 valid language is defined
+            Object.keys(configLanguages).filter(
+              (key) => key !== 'allLanguages' && configLanguages[key].name
+            ).length > 1 && (
```

Comment:
This filter can probably be removed as its made redundant by the `find` above. The length of the Object.keys should do

Reply:
Hm I left the filter in to remove the 'allLanguages' key from the count.

Reply:
Ah in that case let's get rid of the `find` then right? Since the `filter` length will never be `> 1` if the find fails

Reply:
updated in https://github.com/opentripplanner/otp-react-redux/pull/525/commits/83f2573ec464312673ee9a1317c33f66dce53d11